<a href="https://colab.research.google.com/github/GustavoAdolf0/data_contest/blob/main/titanic3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

!pip install ipython-autotime
%load_ext autotime#Proyecto Titanic


se usara ML para determinar que personas sobreviven 

In [ ]:
!pip install ipython-autotime
%load_ext autotime

import pandas as pd
import numpy as np

path= "drive/MyDrive/titanic/"

time: 3.97 ms (started: 2021-11-07 09:47:51 +00:00)


In [ ]:
#data from kaggle

test = pd.read_csv(path+"test.csv",low_memory=False)
train = pd.read_csv(path+"train.csv", low_memory=False)

gender = pd.read_csv(path+"gender_submission.csv", low_memory=False)

time: 532 ms (started: 2021-11-07 09:47:51 +00:00)


In [ ]:
# explorar train

print(train.dtypes)

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object
time: 11 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
train.head()

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

time: 48.7 ms (started: 2021-11-07 09:47:52 +00:00)


![](https://drive.google.com/uc?export=view&id=1DuvAdG2cFy0lPLpES4MyyU1wky9aWRXu)


In [ ]:
len(train), len(test)

(891, 418)

time: 4.27 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
train.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

time: 10.4 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
train.head(5)

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

time: 34.6 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
train.loc[train['Sex'] =="male", 'Sex_aux'] = 6
train.loc[train['Sex'] =="female", 'Sex_aux'] = 7

train["aux1"] = train["Pclass"]*train["Sex_aux"]

time: 29.5 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
train

PassengerId  Survived  Pclass  ... Embarked Sex_aux  aux1
0              1         0       3  ...        S     6.0  18.0
1              2         1       1  ...        C     7.0   7.0
2              3         1       3  ...        S     7.0  21.0
3              4         1       1  ...        S     7.0   7.0
4              5         0       3  ...        S     6.0  18.0
..           ...       ...     ...  ...      ...     ...   ...
886          887         0       2  ...        S     6.0  12.0
887          888         1       1  ...        S     7.0   7.0
888          889         0       3  ...        S     7.0  21.0
889          890         1       1  ...        C     6.0   6.0
890          891         0       3  ...        Q     6.0  18.0

[891 rows x 14 columns]

time: 55.4 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
t_class= pd.read_excel(path+ "titanic_class.xlsx")
t_class

aux   clasee
0    6  clase_1
1    7  clase_2
2   12  clase_3
3   14  clase_4
4   18  clase_5
5   21  clase_6

time: 240 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
print(t_class.dtypes)

aux        int64
clasee    object
dtype: object
time: 2.41 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
print(train.dtypes)

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
Sex_aux        float64
aux1           float64
dtype: object
time: 14.1 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
t_class['aux'] = t_class['aux'].astype(float)
t_class.rename(columns={"aux":"aux1"}, inplace=True)

time: 3.69 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
train= pd.merge(train, t_class, on="aux1", how= "left")

time: 22.9 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
train

PassengerId  Survived  Pclass  ... Sex_aux  aux1   clasee
0              1         0       3  ...     6.0  18.0  clase_5
1              2         1       1  ...     7.0   7.0  clase_2
2              3         1       3  ...     7.0  21.0  clase_6
3              4         1       1  ...     7.0   7.0  clase_2
4              5         0       3  ...     6.0  18.0  clase_5
..           ...       ...     ...  ...     ...   ...      ...
886          887         0       2  ...     6.0  12.0  clase_3
887          888         1       1  ...     7.0   7.0  clase_2
888          889         0       3  ...     7.0  21.0  clase_6
889          890         1       1  ...     6.0   6.0  clase_1
890          891         0       3  ...     6.0  18.0  clase_5

[891 rows x 15 columns]

time: 53.9 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
clases_aux= train.groupby("clasee").mean()
clases_aux.reset_index(inplace=True)
clases_aux=clases_aux[["clasee","Age"]]
clases_aux.rename(columns={"Age":"Age_avg"}, inplace=True)
clases_aux["Age_avg"] = round(clases_aux["Age_avg"],0)
clases_aux

clasee  Age_avg
0  clase_1     41.0
1  clase_2     35.0
2  clase_3     31.0
3  clase_4     29.0
4  clase_5     27.0
5  clase_6     22.0

time: 30.7 ms (started: 2021-11-07 09:47:52 +00:00)


In [ ]:
train= pd.merge(train, clases_aux, on="clasee", how= "left")

time: 12.2 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train.Age.fillna(train.Age_avg, inplace= True)

train.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
Sex_aux          0
aux1             0
clasee           0
Age_avg          0
dtype: int64

time: 20.6 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
emb_aux= train.groupby("Embarked").count()

emb_aux

PassengerId  Survived  Pclass  Name  ...  Sex_aux  aux1  clasee  Age_avg
Embarked                                       ...                                
C                 168       168     168   168  ...      168   168     168      168
Q                  77        77      77    77  ...       77    77      77       77
S                 644       644     644   644  ...      644   644     644      644

[3 rows x 15 columns]

time: 30.2 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train.Embarked.fillna("S", inplace=True)

time: 4.65 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         0
Sex_aux          0
aux1             0
clasee           0
Age_avg          0
dtype: int64

time: 10.4 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
#sólo quedan los na de cabinm dado que son muchos (77%) se eliminara la columna
train.drop(['Cabin'], axis=1, inplace=True)

time: 4.79 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train.head()

PassengerId  Survived  Pclass  ...  aux1   clasee  Age_avg
0            1         0       3  ...  18.0  clase_5     27.0
1            2         1       1  ...   7.0  clase_2     35.0
2            3         1       3  ...  21.0  clase_6     22.0
3            4         1       1  ...   7.0  clase_2     35.0
4            5         0       3  ...  18.0  clase_5     27.0

[5 rows x 15 columns]

time: 37.6 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
#eliminar las col auxiliares
train.drop(["Sex_aux","aux1" , "Age_avg" ], axis=1, inplace=True)
train.head()

PassengerId  Survived  Pclass  ...     Fare Embarked   clasee
0            1         0       3  ...   7.2500        S  clase_5
1            2         1       1  ...  71.2833        C  clase_2
2            3         1       3  ...   7.9250        S  clase_6
3            4         1       1  ...  53.1000        S  clase_2
4            5         0       3  ...   8.0500        S  clase_5

[5 rows x 12 columns]

time: 30.5 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
#algunas categorias se pasaran a one hot encoder P Class , embarked y clasee
one_hot= pd.get_dummies(train["Pclass"], prefix="Pclass")
train=train.join(one_hot)

one_hot= pd.get_dummies(train["Embarked"], prefix="Embarked")
train=train.join(one_hot)

one_hot= pd.get_dummies(train["clasee"])
train=train.join(one_hot)

one_hot= pd.get_dummies(train["Sex"])
train=train.join(one_hot)

train.drop(['Embarked',"Pclass","clasee", "Sex"], axis=1, inplace=True)
train.head()

PassengerId  Survived  ... female  male
0            1         0  ...      0     1
1            2         1  ...      1     0
2            3         1  ...      1     0
3            4         1  ...      1     0
4            5         0  ...      0     1

[5 rows x 22 columns]

time: 77.8 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train

PassengerId  Survived  ... female  male
0              1         0  ...      0     1
1              2         1  ...      1     0
2              3         1  ...      1     0
3              4         1  ...      1     0
4              5         0  ...      0     1
..           ...       ...  ...    ...   ...
886          887         0  ...      0     1
887          888         1  ...      1     0
888          889         0  ...      1     0
889          890         1  ...      0     1
890          891         0  ...      0     1

[891 rows x 22 columns]

time: 53.2 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
miss = (train[train["Name"].str.contains("Miss")]).copy(deep=True)
miss["miss"] = 1
miss = miss[["PassengerId","miss"]]

mister = (train[train["Name"].str.contains("Mr.")]).copy(deep=True)
mister["mister"] = 1
mister = mister[["PassengerId","mister"]]

misteres = (train[train["Name"].str.contains("Mrs.")]).copy(deep=True)
misteres["misteres"] = 1
misteres = misteres[["PassengerId","misteres"]]

rev = (train[train["Name"].str.contains("Rev.")]).copy(deep=True)
rev["rev"] = 1
rev = rev[["PassengerId","rev"]]

rev

PassengerId  rev
149          150    1
150          151    1
249          250    1
626          627    1
848          849    1
886          887    1

time: 108 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train = pd.merge(train, miss, on="PassengerId", how= "left")

train = pd.merge(train, mister, on="PassengerId", how= "left")

train = pd.merge(train, misteres, on="PassengerId", how= "left")

train = pd.merge(train, rev, on="PassengerId", how= "left")




time: 24.1 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
n_train=train.copy(deep=True)
n_train.fillna(0, inplace= True)
n_train["resumen"]= n_train["miss"] + n_train["mister"] + n_train["misteres"] + n_train["rev"]

n_train = n_train[["PassengerId","resumen"]]

null_data = n_train[n_train.isnull().any(axis=1)]
null_data

Empty DataFrame
Columns: [PassengerId, resumen]
Index: []

time: 18.2 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
#no existe nan
train.fillna(0, inplace= True)

time: 4.27 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
#se genera casados (mr y mrs) y solteros (miss)


time: 1.03 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
#ticket y name no poseen información agrupable
train.drop(['Name',"Ticket"], axis=1, inplace=True)

train["fam"]= train["SibSp"] + train["Parch"]

train.head()


PassengerId  Survived   Age  SibSp  Parch  ...  miss  mister  misteres  rev  fam
0            1         0  22.0      1      0  ...   0.0     1.0       0.0  0.0    1
1            2         1  38.0      1      0  ...   0.0     1.0       1.0  0.0    1
2            3         1  26.0      0      0  ...   1.0     0.0       0.0  0.0    0
3            4         1  35.0      1      0  ...   0.0     1.0       1.0  0.0    1
4            5         0  35.0      0      0  ...   0.0     1.0       0.0  0.0    0

[5 rows x 25 columns]

time: 56.3 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
edad_class= pd.read_excel(path+"clases_edad.xlsx")
edad_class.rename(columns={"edad":"Age"}, inplace=True)
edad_class['riesgo'] = edad_class['riesgo'].astype(int)
print(edad_class.dtypes)

Age       int64
riesgo    int64
dtype: object
time: 182 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train['Age'] = train['Age'].astype(int)

time: 2.86 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train = pd.merge(train, edad_class, on="Age", how= "left")


time: 9.62 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train

PassengerId  Survived  Age  SibSp  ...  misteres  rev  fam  riesgo
0              1         0   22      1  ...       0.0  0.0    1       6
1              2         1   38      1  ...       1.0  0.0    1       6
2              3         1   26      0  ...       0.0  0.0    0       6
3              4         1   35      1  ...       1.0  0.0    1       6
4              5         0   35      0  ...       0.0  0.0    0       6
..           ...       ...  ...    ...  ...       ...  ...  ...     ...
886          887         0   27      0  ...       0.0  1.0    0       6
887          888         1   19      0  ...       0.0  0.0    0       6
888          889         0   22      1  ...       0.0  0.0    3       6
889          890         1   26      0  ...       0.0  0.0    0       6
890          891         0   32      0  ...       0.0  0.0    0       6

[891 rows x 26 columns]

time: 56.9 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train.isna().sum()

PassengerId    0
Survived       0
Age            0
SibSp          0
Parch          0
Fare           0
Pclass_1       0
Pclass_2       0
Pclass_3       0
Embarked_C     0
Embarked_Q     0
Embarked_S     0
clase_1        0
clase_2        0
clase_3        0
clase_4        0
clase_5        0
clase_6        0
female         0
male           0
miss           0
mister         0
misteres       0
rev            0
fam            0
riesgo         0
dtype: int64

time: 8.94 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
null_data = train[train.isnull().any(axis=1)]
null_data

Empty DataFrame
Columns: [PassengerId, Survived, Age, SibSp, Parch, Fare, Pclass_1, Pclass_2, Pclass_3, Embarked_C, Embarked_Q, Embarked_S, clase_1, clase_2, clase_3, clase_4, clase_5, clase_6, female, male, miss, mister, misteres, rev, fam, riesgo]
Index: []

time: 20.4 ms (started: 2021-11-07 09:47:53 +00:00)


In [ ]:
train.set_index("PassengerId", inplace=True)
train.head()

Survived  Age  SibSp  Parch  ...  misteres  rev  fam  riesgo
PassengerId                               ...                            
1                   0   22      1      0  ...       0.0  0.0    1       6
2                   1   38      1      0  ...       1.0  0.0    1       6
3                   1   26      0      0  ...       0.0  0.0    0       6
4                   1   35      1      0  ...       1.0  0.0    1       6
5                   0   35      0      0  ...       0.0  0.0    0       6

[5 rows x 25 columns]

time: 42.9 ms (started: 2021-11-07 09:47:54 +00:00)


In [ ]:
#modelos a usar

# Models from Scikit-Learn
from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import RandomForestClassifier

# Model Evaluations
from sklearn.model_selection import train_test_split, cross_val_score
from sklearn.model_selection import RandomizedSearchCV, GridSearchCV
from sklearn.metrics import confusion_matrix, classification_report
from sklearn.metrics import precision_score, recall_score, f1_score
from sklearn.metrics import plot_roc_curve

time: 1.01 s (started: 2021-11-07 09:47:54 +00:00)


In [ ]:
# Split data into X and y
X = train.drop("Survived", axis=1)

y = train["Survived"]


# Split data into train and test sets
np.random.seed(42)

# Split into train & test set
X_train, X_test, y_train, y_test = train_test_split(X,
                                                    y,
                                                    test_size=0.1)

time: 13.6 ms (started: 2021-11-07 09:47:55 +00:00)


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
time: 3.38 ms (started: 2021-11-07 09:47:55 +00:00)


In [ ]:
X

Age  SibSp  Parch     Fare  ...  misteres  rev  fam  riesgo
PassengerId                              ...                            
1             22      1      0   7.2500  ...       0.0  0.0    1       6
2             38      1      0  71.2833  ...       1.0  0.0    1       6
3             26      0      0   7.9250  ...       0.0  0.0    0       6
4             35      1      0  53.1000  ...       1.0  0.0    1       6
5             35      0      0   8.0500  ...       0.0  0.0    0       6
...          ...    ...    ...      ...  ...       ...  ...  ...     ...
887           27      0      0  13.0000  ...       0.0  1.0    0       6
888           19      0      0  30.0000  ...       0.0  0.0    0       6
889           22      1      2  23.4500  ...       0.0  0.0    3       6
890           26      0      0  30.0000  ...       0.0  0.0    0       6
891           32      0      0   7.7500  ...       0.0  0.0    0       6

[891 rows x 24 columns]

time: 60.3 ms (started: 2021-11-07 09:47:55 +00:00)


In [ ]:
y_train

PassengerId
166    1
542    0
626    0
389    0
77     0
      ..
107    1
271    0
861    0
436    1
103    0
Name: Survived, Length: 801, dtype: int64

time: 6.4 ms (started: 2021-11-07 09:47:55 +00:00)


In [ ]:
# Put models in a dictionary
models = {"Logistic Regression": LogisticRegression(),
          "KNN": KNeighborsClassifier(),
          "Random Forest": RandomForestClassifier()}

# Create a function to fit and score models
def fit_and_score(models, X_train, X_test, y_train, y_test):
    """
    Fits and evaluates given machine learning models.
    models : a dict of differetn Scikit-Learn machine learning models
    X_train : training data (no labels)
    X_test : testing data (no labels)
    y_train : training labels
    y_test : test labels
    """
    # Set random seed
    np.random.seed(42)
    # Make a dictionary to keep model scores
    model_scores = {}
    # Loop through models
    for name, model in models.items():
        # Fit the model to the data
        model.fit(X_train, y_train)
        # Evaluate the model and append its score to model_scores
        model_scores[name] = model.score(X_test, y_test)
    return model_scores

time: 13 ms (started: 2021-11-07 09:47:55 +00:00)


In [ ]:
model_scores = fit_and_score(models=models,
                             X_train=X_train,
                             X_test=X_test,
                             y_train=y_train,
                             y_test=y_test)

model_scores

/usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


{'KNN': 0.7444444444444445,
 'Logistic Regression': 0.8333333333333334,
 'Random Forest': 0.8444444444444444}

time: 318 ms (started: 2021-11-07 09:47:55 +00:00)


n_estimators : This is the number of trees you want to build before taking the maximum voting or averages of predictions. Higher number of trees give you better performance but makes your code slower.

max_depth es la profundidad del último nodo de desición creado, el nodo raiz es la profundidad uno, en el tree 1, es de 3 dado que el pultimo en generar leaf es la en la tercera corrida hacia abajo.

![](https://drive.google.com/uc?export=view&id=1EUGRJjmSUtoK8aAe4_Gi6aOt-6ULhERo)


min_samples_split es la cantidad mínima de observaciones necesarias para generar una divisón en el nodo

min_sample_leaf es la cantidad mínima de observaciones que pueden entrar en un nodo


![](https://drive.google.com/uc?export=view&id=1ymIGEAujGesx7smtFn30AoVI6LNbXaUa)


In [ ]:
# Create a hyperparameter grid for RandomForestClassifier
rf_grid = {"n_estimators": np.arange(10, 2500, 10),
           "max_depth": [ None, 10, 15, 20],
           "min_samples_split": np.arange(2, 20, 2),
           "min_samples_leaf": np.arange(2, 20, 2)}

time: 3.66 ms (started: 2021-11-07 09:47:55 +00:00)


CV es el hyperparamter para cros validation el modelo selecciona 1 de K para ser usado para entrenamiento y el k-esimo para usarlo en validación, de esta forma se evita el overfitting

![](https://drive.google.com/uc?export=view&id=1cOfpIUcgy5L38z9M9vL42CJd6xdAkUMD)






In [ ]:
# Setup random seed
np.random.seed(42)

# Setup random hyperparameter search for RandomForestClassifier
rs_rf = RandomizedSearchCV(RandomForestClassifier(), 
                           param_distributions=rf_grid,
                           cv=5,
                           n_iter=500,
                           verbose=True)

# Fit random hyperparameter search model for RandomForestClassifier()
rs_rf.fit(X_train, y_train)

Fitting 5 folds for each of 500 candidates, totalling 2500 fits


[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
[Parallel(n_jobs=1)]: Done 2500 out of 2500 | elapsed: 85.4min finished


RandomizedSearchCV(cv=5, error_score=nan,
                   estimator=RandomForestClassifier(bootstrap=True,
                                                    ccp_alpha=0.0,
                                                    class_weight=None,
                                                    criterion='gini',
                                                    max_depth=None,
                                                    max_features='auto',
                                                    max_leaf_nodes=None,
                                                    max_samples=None,
                                                    min_impurity_decrease=0.0,
                                                    min_impurity_split=None,
                                                    min_samples_leaf=1,
                                                    min_samples_split=2,
                                                    min_weight_fraction_leaf=0.0,
               

time: 1h 25min 27s (started: 2021-11-07 09:47:55 +00:00)


In [ ]:
# Find the best hyperparameters
rs_rf.best_params_

{'max_depth': 10,
 'min_samples_leaf': 1,
 'min_samples_split': 10,
 'n_estimators': 780}

time: 12.7 ms (started: 2021-11-07 11:15:23 +00:00)


{'max_depth': 20,
 
 'min_samples_leaf': 5,

 'min_samples_split': 6,

 rs_rf.score(X_test, y_test)

 0.8222

{'max_depth': 20,
 
 'min_samples_leaf': 1,

 'min_samples_split': 14,

 'n_estimators': 1520}
 rs_rf.score(X_test, y_test)
 0.844444

{'max_depth': None,

 'min_samples_leaf': 1,

 'min_samples_split': 10,

 'n_estimators': 10}

rs_rf.score(X_test, y_test)

 0.866666

In [ ]:
# Evaluate the randomized search RandomForestClassifier model
rs_rf.score(X_test, y_test)

0.8444444444444444

time: 98.1 ms (started: 2021-11-07 11:15:31 +00:00)


In [ ]:
def pre_pross(df):

  #this function grab from the file read to ready to ML
  df.loc[df["Sex"] =="male", "Sex_aux"] = 6
  df.loc[df["Sex"] == "female", "Sex_aux"] = 7 #create aux columns to 
  df["aux1"] = df["Pclass"]*df["Sex_aux"] #col aux to use in merge

  df= pd.merge(df, t_class, on="aux1", how= "left") #merge with t_class
  df= pd.merge(df, clases_aux, on="clasee", how= "left") #merge with clases to get age avg
  df.Age.fillna(train.Age_avg, inplace= True)
  df.Embarked.fillna("S", inplace=True)
  df.drop(['Cabin'], axis=1, inplace=True)
  df.drop(["Sex_aux","aux1" , "Age_avg" ], axis=1, inplace=True)

  one_hot= pd.get_dummies(train["Pclass"], prefix="Pclass")
  df=df.join(one_hot)

  one_hot= pd.get_dummies(train["Embarked"], prefix="Embarked")
  df=df.join(one_hot)

  one_hot= pd.get_dummies(train["clasee"])
  df=df.join(one_hot)

  one_hot= pd.get_dummies(train["Sex"])
  df=df.join(one_hot)

  df.drop(['Embarked',"Pclass","clasee", "Sex"], axis=1, inplace=True)

  miss = (df[df["Name"].str.contains("Miss")]).copy(deep=True)
  miss["miss"] = 1
  miss = miss[["PassengerId","miss"]]

  mister = (df[df["Name"].str.contains("Mr.")]).copy(deep=True)
  mister["mister"] = 1
  mister = mister[["PassengerId","mister"]]

  misteres = (df[df["Name"].str.contains("Mrs.")]).copy(deep=True)
  misteres["misteres"] = 1
  misteres = misteres[["PassengerId","misteres"]]

  rev = (df[df["Name"].str.contains("Rev.")]).copy(deep=True)
  rev["rev"] = 1
  rev = rev[["PassengerId","rev"]]

  df = pd.merge(df, miss, on="PassengerId", how= "left")
  df = pd.merge(df, mister, on="PassengerId", how= "left")
  df = pd.merge(df, misteres, on="PassengerId", how= "left")
  df = pd.merge(df, rev, on="PassengerId", how= "left")
  df.fillna(0, inplace= True)

  df.drop(['Name',"Ticket"], axis=1, inplace=True)
  df["fam"]= df["SibSp"] + df["Parch"]
  
  df = pd.merge(df, edad_class, on="Age", how= "left")

  df.set_index("PassengerId", inplace=True)
  return df
   



In [ ]:
pre_pross(test)

NameError: ignored